### PR TITLE
[Port List Features] List Feature fixes

### PR DESCRIPTION
**Pull request by juliaroldi**
Fix content model list features: 
1. Maintain list feature when use auto list 
maintain continue list

2. Use delete key to delete list 
delete-to-delete-key

3. Add Enter key features for list 
* Maintain list after press key in a list item.
Enter-key-to-mantain-list
* Enter key to delete a empty list item 
enter-to-delete

4. Set list paragraph to be non implicit, so the cursor can be focused on the paragraph. 
DefaultStyleSafariList

5. Enter + Shift, creates a new line on the list item 
shift+EnterList

### CODE REVIEW

File: packages-content-model/roosterjs-content-model-api/lib/modelApi/list/findListItemsInSameThread.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import type {
 } from 'roosterjs-content-model-types';
 
 /**
- * @internal
+ * Search for all list items in the same thread as the current list item
```
Comment (JiuqingSong) [inline packages-content-model/roosterjs-content-model-api/lib/modelApi/list/findListItemsInSameThread.ts:8]: Please add comments for parameters

File: packages-content-model/roosterjs-content-model-plugins/lib/edit/inputSteps/handleEnterOnList.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    DeleteSelectionStep,
+    InsertPoint,
+    ValidDeleteSelectionContext,
+} from 'roosterjs-content-model-types';
+
+/**
+ * @internal
+ */
+export const handleEnterOnList: DeleteSelectionStep = context => {
+    if (context.deleteResult == 'notDeleted') {
```
[the remark this answers is not in the document]
Comment (juliaroldi) [reply packages-content-model/roosterjs-content-model-plugins/lib/edit/inputSteps/handleEnterOnList.ts]: Do you mean like paragraph with not collapsed range? I added some code to handle this scenario. 
handleRangeList

[the remark this answers is not in the document]
Comment (JiuqingSong) [reply packages-content-model/roosterjs-content-model-plugins/lib/edit/inputSteps/handleEnterOnList.ts]: If I press ENTER, it should not only delete the selected text, but also create a new list item, as if I press DELETE+ENTER

[the remark this answers is not in the document]
Comment (juliaroldi) [reply packages-content-model/roosterjs-content-model-plugins/lib/edit/inputSteps/handleEnterOnList.ts]: I changed the behavior to work like this
enterOnExpandedSelection

**Comment by JiuqingSong** [comment]
Try restore the snapshot below, the press Enter.
```html
<ol style="margin-block: 0px; list-style-type: decimal;" data-editing-info="{&quot;orderedStyleType&quot;:1,&quot;unorderedStyleType&quot;:1}"><li style="list-style-type: &quot;1. &quot;;">aaaa</li></ol><div>bbb</div><div><ol style="margin-block: 0px;" data-editing-info="{&quot;orderedStyleType&quot;:1,&quot;unorderedStyleType&quot;:1}"><li style="list-style-type: &quot;1. &quot;;"><span>cccc</span></li></ol></div><!--{"type":"range","start":[0,0,0,2],"end":[2,0,0,0,0,2],"isDarkMode":false}-->
```

Expected:
image

Actual:
image

**Comment by JiuqingSong** [comment]
```html
<div><ol data-listchain="__List_Chain_3" style="margin-block: 0px;" data-editing-info="{&quot;orderedStyleType&quot;:1,&quot;unorderedStyleType&quot;:1}"><li style="list-style-type: &quot;1. &quot;;"><span>aaaa<br><br>bbbb</span></li><li style="list-style-type: &quot;2. &quot;;"><span>dddd</span></li></ol></div><!--{"type":"range","start":[0,0,0,0,2],"end":[0,0,0,0,2],"isDarkMode":false}-->
```
Press Enter key, it should create a new list item on the new line, but it actually create list number on the same line

**Comment by JiuqingSong** [comment]
The above two comments are some bugs that I found we I try to work on the Enter key but not solved yet. So hope you can solve them together.

Another issue is, if I press Shift+Enter, it should create a new line without new list number.

**Review by JiuqingSong — APPROVED** [review]

**Review by JiuqingSong — APPROVED** [review]

**Comment by JiuqingSong** [comment]
I can still repro the issue below, is it planned to be fixed later?

enteronlist

**Review by JiuqingSong — APPROVED** [review]

**Pull request merged**